src/main/java/seedu/toluist/ui/view/TaskUiView.java

```diff
@@ -95,6 +101,38 @@ protected void viewDidMount() {
         }
         if (isTaskWithDeadline || task.isEvent()) {
             clockIcon.setImage(AppUtil.getImage(IMAGE_PATH_CLOCK_ICON));
+        } else {
+            date.setVisible(false);
+            date.setManaged(false);
+            clockIcon.setVisible(false);
+            clockIcon.setManaged(false);
+        }
+
+        if (isRecurring) {
+            recurringIcon.setImage(AppUtil.getImage(IMAGE_PATH_RECURRING_ICON));
+            if (isFloatingTask) {
+                recurringDate.setText(DateTimeFormatterUtil.formatRecurringFloatingTask(
+                        task.getRecurringEndDateTime(),
+                        task.getRecurringFrequency()));
+            } else if (isTaskWithDeadline) {
+                recurringDate.setText(
+                        DateTimeFormatterUtil.formatRecurringTaskDeadline(
+                                task.getEndDateTime(),
+                                task.getRecurringEndDateTime(),
+                                task.getRecurringFrequency()));
+            } else if (isEvent) {
+                recurringDate.setText(
+                        DateTimeFormatterUtil.formatRecurringEvent(
```

Comment:
not too sure if this is the correct way to indent based on 2103 coding guideline